docs: fix `SpyDirective` in `lifecycle-hooks` docs example to use one ID per instance

gkalpak opened the pull request:
Previously, the `SpyDirective` in the `lifecycle-hooks` docs example would use a different ID when logging `onInit` and when logging `onDestroy` for the same instance, making it impossible to associate the two calls. This was not helpful and came in constrast with how the directive was described in the corresponding guide and shown in the accompanying `spy-directive.gif` image.

This PR fixes the logic of the `SpyDirective` class to use the same ID for all log operations of an instance.

Partially addresses #40193.

##
This PR also slightly refactors the `lifecycle-hooks` docs example to be a little easier to follow. See individual commits for details.

mary-poppins commented:
You can preview 54cd283 at https://pr40208-54cd283.ngbuilds.io/.

gkalpak commented:
For some reason, the CircleCI jobs do not show up in the GitHub checks list, but they _did_ run and pass: https://app.circleci.com/pipelines/github/angular/angular/26306/workflows/02685daa-f5db-44cc-b3ae-1ac4e73fc8ab

(In fact, closing and reopening the PR makes the jobs appear in the checks list for a couple of seconds and then they disappear again 🤷‍♂️)

AlexElin commented on aio/content/examples/lifecycle-hooks/src/app/app.component.html:
```diff
@@ -32,6 +28,10 @@ <h1>Component Lifecycle Hooks</h1>
 <after-content-parent></after-content-parent>
 <a href="#top">back to top</a>
 
+<a id="spy"></a>
```
Here you've placed the component at the same position (penultimate) that it has in the upper menu in the example's page.
But in the [Lifecycle example set](https://angular.io/guide/lifecycle-hooks#lifecycle-example-set) and in the [General examples](https://angular.io/guide/lifecycle-hooks#general-examples) it goes at the second position after the peek-a-boo component
So I think the `spy-parent` component was placed correctly but it has had _wrong position in the menu_.
It should be second according to the docs content.

gkalpak replied:
Good point. I amended the commit to move the link instead.
Thx for catching this 👍

gkalpak committed:
docs: make the `spy-directive` docregion (in `lifecycle-hooks` example) easier to follow

Previously, the docregion code referenced a `nextId` variable that was
not shown in the code, which was confusing for the reader.

This commit makes the declaration of the `nextId` variable part of the
docregion, so it is clear to the reader where it comes from and how it
is initialized.
This commit also removes the `logIt()` helper method, which didn't seem
to add value and calls `logger.log()` directly instead.

gkalpak committed:
docs: fix `SpyDirective` in `lifecycle-hooks` docs example to use one ID per instance

Previously, the `SpyDirective` in the `lifecycle-hooks` docs example
would use a different ID when logging `onInit` and when logging
`onDestroy` for the same instance, making it impossible to associate the
two calls. This was not helpful and came in constrast with how the
directive was described in the corresponding guide and shown in the
accompanying `spy-directive.gif` image.

This commit fixes the logic of the `SpyDirective` class to use the same
ID for all log operations of an instance.

Partially addresses #40193.

gkalpak committed:
docs: fix the order of links in `lifecycle-hooks` docs example

This commit aligns the order of the links to the various sections of the
`lifecycle-hooks` docs example with the order in which the sections
appear in the template (which also coincides with the order in which
they are presented/discussed in the corresponding guide).

mary-poppins commented:
You can preview a30d655 at https://pr40208-a30d655.ngbuilds.io/.

jessicajaniuk reviewed (approved):
LGTM
